compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -338,3 +372,59 @@ pub fn create_compressed_metadata_file(
 
     file.write().unwrap()
 }
+
+/// * Xcoff - AIX target does not support arbitrary section names, the metadata
```

Comment:
The comment of mechanisum to preserve the `.info` section is incomplete. Maybe
```
            // On AIX, custom sections are merged into predefined sections, so                                                                                
            // custom .rustc section is not preserved during linking.                                                                                         
            // For this reason, we store metadata in predefined .info section,                                                                                
            // and define a symbol to reference the metadata.                                                                                                 
            // To preserve metadata during linking on AIX, we have to                                                                                         
            // 1. Create a empty .text section, a empty .data section.                                                                                        
            // 2. Define empty symbol named `symbol_name` inside .data section.                                                                               
            // 3. Define a symbol named `aix_rust_metadata_symbol_name` referencing                                                                           
            //    data inside .info section.                                                                                                                  
            // From XCOFF's view, (2) creates a csect entry in the symbol table,                                                                              
            // the symbol created by (3) is a info symbol for the preceding csect.                                                                            
            // Thus two symbols are preserved during linking and we can use the second                                                                        
            // symbol to reference the metadata.
```

Reply:
Nice comments, thanks.